misc-utils/lastlog2.c

```diff
@@ -0,0 +1,332 @@
+	fputs (_("  -v, --version         Print version number and exit\n"), output);
+	fputs ("\n", output);
+	exit (retval);
+}
+
+/* Check if an user exists on the system.
+   If yes, return 0, else return -1. */
+static int
+check_user (const char *name)
+{
+	if (getpwnam (name) == NULL)
+		return -1;
+	return 0;
+}
+
+int
+main (int argc, char **argv)
+{
+	struct option const longopts[] = {
+		{"before",   required_argument, NULL, 'b'},
+		{"clear",    no_argument,       NULL, 'C'},
+		{"database", required_argument, NULL, 'd'},
+		{"help",     no_argument,       NULL, 'h'},
+		{"import",   required_argument, NULL, 'i'},
+		{"rename",   required_argument, NULL, 'r'},
+		{"service",  no_argument,       NULL, 's'},
+		{"set",      no_argument,       NULL, 'S'},
+		{"time",     required_argument, NULL, 't'},
+		{"user",     required_argument, NULL, 'u'},
+		{"version",  no_argument,       NULL, 'v'},
+		{NULL, 0, NULL, '\0'}
+	};
+	char *error = NULL;
+	int Cflg = 0;
+	int iflg = 0;
+	int rflg = 0;
+	int Sflg = 0;
+	int uflg = 0;
+	const char *user = NULL;
+	const char *newname = NULL;
+	const char *lastlog_file = NULL;
+	int c;
+
+	while ((c = getopt_long (argc, argv, "b:Cd:hi:r:sSt:u:v", longopts, NULL)) != -1) {
+		switch (c) {
+		case 'b': /* before DAYS; Print only records older than DAYS */
+			{
+				unsigned long days;
+				errno = 0;
+				days = strtoul_or_err(optarg, _("Cannot parse days"));
+				b_days = (time_t) days * (24L*3600L) /* seconds/DAY */;
+				bflg = 1;
+			}
+			break;
+		case 'C': /* clear; Clear record of a user (requires -u) */
+			Cflg = 1;
+			break;
+		case 'd': /* database <FILE>;   Use FILE as lastlog2 database */
+			lastlog2_path = optarg;
+			break;
+		case 'h': /* help; Display this help message and exit */
+			usage (EXIT_SUCCESS);
+			break;
+		case 'i': /* import <FILE>; Import data from old lastlog file */
+			lastlog_file = optarg;
+			iflg = 1;
+			break;
+		case 'r': /* rename <NEWNAME>; Rename existing user to NEWNAME (requires -u) */
+			rflg = 1;
+			newname = optarg;
+			break;
+		case 's': /* service; Display PAM service */
+			sflg = 1;
+			break;
+		case 'S': /* set; Set lastlog record to current time (requires -u) */
+			/* Set lastlog record of a user to the current time. */
+			Sflg = 1;
+			break;
+		case 't': /* time <DAYS>; Print only lastlog records more recent than DAYS */
+			{
+				unsigned long days;
+				errno = 0;
+				days = strtoul_or_err(optarg, _("Cannot parse days"));
+				t_days = (time_t) days * (24L*3600L) /* seconds/DAY */;
+				tflg = 1;
+			}
+			break;
+		case 'u': /* user <LOGIN>; Print lastlog record of the specified LOGIN */
+			uflg = 1;
+			user = optarg;
+			break;
+		case 'v': /* version; Print version number and exit */
+			print_version(EXIT_SUCCESS);
+			break;
+		default:
+			usage (EXIT_FAILURE);
+			break;
+		}
+	}
+
+	if (argc > optind) {
+		fprintf (stderr, _("Unexpected argument: %s\n"), argv[optind]);
+		usage (EXIT_FAILURE);
```

Comment:
We do not use usage() on error anymore in util-linux. Please print the error by err().

 (Or, in soma cases, we use warn() and errtryhelp(EXIT_FAILURE)).

Reply:
I'll also suggest not using space between the function name and '('.

Reply:
I have used errx(). Is this OK ?